packages/app-desktop/gui/NoteEditor/utils/resourceHandling.ts

```diff
@@ -125,3 +126,40 @@ export async function handlePasteEvent(event: any) {
 	}
 	return output;
 }
+
+export async function processPastedHtml(html: string) {
+	const allImageUrls: string[] = [];
+	const mappedResources: Record<string, string> = {};
+
+	htmlUtils.replaceImageUrls(html, (src: string) => {
+		allImageUrls.push(src);
+	});
+
+	for (const imageSrc of allImageUrls) {
+		if (!mappedResources[imageSrc]) {
+			try {
+				if (imageSrc.startsWith('file')) {
+					const imageFilePath = imageSrc.substr(7);
+
+					if (imageFilePath.startsWith(Setting.value('resourceDir'))) {
+						mappedResources[imageSrc] = imageSrc;
+					} else {
+						const createdResource = await shim.createResourceFromPath(imageFilePath);
+						mappedResources[imageSrc] = `file://${Resource.fullPath(createdResource)}`;
+					}
+				} else {
+					const filePath = `${Setting.value('tempDir')}/${md5(Date.now())}`;
+					await shim.fetchBlob(imageSrc, { path: filePath });
+					const createdResource = await shim.createResourceFromPath(filePath);
+					mappedResources[imageSrc] = `file://${Resource.fullPath(createdResource)}`;
+				}
+			} catch (err) {
+				mappedResources[imageSrc] = imageSrc;
```

Comment:
Please log a warning in this case. Create a new logger `const logger = Logger.create('resourceHandling');` then `logger.warn('......')`

Reply:
done

Reply:
Check how and in particular **where** the logger is declared in other files.

Reply:
Actually, in this file itself, the logger is used using `reg`, as follows.

```javascript
reg.logger().error(error);
```

Reply:
Should I use it in this way?

Reply:
I'll fix it.